### PR TITLE
[14.0] Add shopinvader_customer_validate

### PR DESCRIPTION
**Pull request by simahawk**
* move out customer validation from core
* refactor validation using multiple states

replaces #1012

### CODE REVIEW

File: shopinvader/models/shopinvader_backend.py [hunk 1]
```diff
@@ -227,6 +188,28 @@ class ShopinvaderBackend(models.Model):
     customer_default_role = fields.Char(
         compute="_compute_customer_default_role",
     )
+    salesman_notify_create = fields.Selection(
```
Comment (simahawk) [inline shopinvader/models/shopinvader_backend.py:191]: FTR these are not related to validation, I just moved the fields to bottom.

**Review by ivantodorovich — APPROVED** [review]
